bigquery-storage/api/BigQueryStorage.Samples.Tests/BigQueryStorageFixture.cs

```diff
@@ -0,0 +1,121 @@
+
+    internal Dictionary<long, (long, string)> TestData { get; }
+
+    public BigQueryStorageFixture()
+    {
+        ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        if (string.IsNullOrWhiteSpace(ProjectId))
+        {
+            throw new Exception("Missing GOOGLE_PROJECT_ID. You need to set the Environment variable 'GOOGLE_PROJECT_ID' with your Google Cloud Project's project id.");
+        }
+
+        string credentials = Environment.GetEnvironmentVariable("GOOGLE_APPLICATION_CREDENTIALS");
+        if (string.IsNullOrWhiteSpace(credentials))
+        {
+            throw new Exception("Missing GOOGLE_APPLICATION_CREDENTIALS. You need to set the Environment variable 'GOOGLE_APPLICATION_CREDENTIALS' with your Google Cloud Project's credential json.");
+        }
+
+        if(!System.IO.File.Exists(credentials))
+        {
+            throw new Exception("Invalid GOOGLE_APPLICATION_CREDENTIALS. The file path specified in the Environment variable 'GOOGLE_APPLICATION_CREDENTIALS' does not exist.");
+        }
+
+        TestData = new Dictionary<long, (long, string)>();
+        PopulateTestData();
+    }
+
+    /// <summary>
+    /// This dictionary is used to check the inserted data. 
+    /// For keeping sample code clean and independent of external data (like in Python sample), 
+    /// please ensure to use the same test data in samples.
+    /// </summary>
+    internal void PopulateTestData()
+    {
+        TestData.Add(1, (1, "Alice"));
+        TestData.Add(2, (2, "Bob"));
+        TestData.Add(3, (3, "Charles"));
+    }
+
+    /// <summary>
+    /// Ensures the table with given name exists in the BigQuery Dataset
+    /// </summary>
+    internal void EnsureDataSetAndTableExists(string tableName)
+    {
+        BigQueryClient client = BigQueryClient.Create(ProjectId);
+        BigQueryDataset dataSet = client.GetOrCreateDataset(DatasetId);
+
+        TableSchema schema = new TableSchemaBuilder()
+        {
+            { "row_num", BigQueryDbType.Int64 },
+            { "customer_name", BigQueryDbType.String }
+        }.Build();
+
+        dataSet.GetOrCreateTable(tableName, schema);
+    }
+
+    /// <summary>
+    ///  Disposes the instance of BigQueryStorageFixture
+    /// </summary>
+    public void Dispose() => DeleteDataSet();
+
+    /// <summary>
+    ///  Gets the table data rows for the given table name in the BigQuery Dataset.
+    /// </summary>
+    /// <returns></returns>
+    internal async Task<PagedEnumerable<TableDataList, BigQueryRow>> GetTableDataAsync(string tableName)
+    {
+        BigQueryClient client = await BigQueryClient.CreateAsync(ProjectId);
+        BigQueryTable table = await client.GetTableAsync(DatasetId, tableName);
+        return table.ListRows();
+    }
+
+    /// <summary>
+    /// Deletes the table with given name in the BigQuery Dataset.
+    /// </summary>
+    internal void DeleteTable(string tableName)
```

Comment:
We delete the dataset on dispose, we don't need to delete tables in between.

Reply:
Yes. It was unused method. Removed it.